x-pack/test/api_integration/apis/ml/results/get_category_definition.ts

```diff
@@ -0,0 +1,184 @@
+import { FtrProviderContext } from '../../../ftr_provider_context';
+import { USER } from '../../../../functional/services/ml/security_common';
+import { COMMON_REQUEST_HEADERS } from '../../../../functional/services/ml/common_api';
+import { Datafeed } from '../../../../../plugins/ml/common/types/anomaly_detection_jobs';
+
+export default ({ getService }: FtrProviderContext) => {
+  const esArchiver = getService('esArchiver');
+  const supertest = getService('supertestWithoutAuth');
+  const ml = getService('ml');
+  const spacesService = getService('spaces');
+
+  const jobIdSpace1 = `sample_logs_${Date.now()}`;
+  const idSpace1 = 'space1';
+  const idSpace2 = 'space2';
+
+  const PARTITION_FIELD_NAME = 'event.dataset';
+  const testJobConfig = {
+    job_id: jobIdSpace1,
+    groups: ['sample_logs', 'bootstrap', 'categorization'],
+    description: "count by mlcategory (message) on 'sample logs' dataset with 15m bucket span",
+    analysis_config: {
+      bucket_span: '15m',
+      categorization_field_name: 'message',
+      per_partition_categorization: { enabled: true, stop_on_warn: true },
+      detectors: [
+        {
+          function: 'count',
+          by_field_name: 'mlcategory',
+          partition_field_name: PARTITION_FIELD_NAME,
+        },
+      ],
+      influencers: ['mlcategory'],
+    },
+    analysis_limits: { model_memory_limit: '26MB' },
+    data_description: { time_field: '@timestamp', time_format: 'epoch_ms' },
+    model_plot_config: { enabled: false, annotations_enabled: true },
+    model_snapshot_retention_days: 10,
+    daily_model_snapshot_retention_after_days: 1,
+    allow_lazy_open: false,
+  };
+  // @ts-expect-error not full interface
+  const testDatafeedConfig: Datafeed = {
+    datafeed_id: `datafeed-${jobIdSpace1}`,
+    indices: ['ft_module_sample_logs'],
+    job_id: jobIdSpace1,
+    query: { bool: { must: [{ match_all: {} }] } },
+  };
+
+  const expectedCategoryDefinition = {
+    categoryId: '1',
+    terms: 'GET HTTP/1.1 Mozilla/5.0 X11 Linux x86_64 rv Gecko/20110421 Firefox/6.0a1',
+    regex:
+      '.*?GET.+?HTTP/1\\.1.+?Mozilla/5\\.0.+?X11.+?Linux.+?x86_64.+?rv.+?Gecko/20110421.+?Firefox/6\\.0a1.*',
+    examples: [
+      '130.246.123.197 - - [2018-07-22T03:26:21.326Z] "GET /beats/metricbeat HTTP/1.1" 200 6850 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+      '130.246.123.197 - - [2018-07-22T03:26:21.326Z] "GET /beats/metricbeat_1 HTTP/1.1" 200 6850 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+      '223.87.60.27 - - [2018-07-22T00:39:02.912Z] "GET /elasticsearch/elasticsearch-6.3.2.deb HTTP/1.1" 200 6219 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+      '223.87.60.27 - - [2018-07-22T00:39:02.912Z] "GET /elasticsearch/elasticsearch-6.3.2.deb_1 HTTP/1.1" 200 6219 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+    ],
+  };
```

Comment:
With the changed assertions, we no longer need the values for terms, regex and examples anymore. SO this could be something like this:
```
  const expectedCategoryDefinition = {
    categoryId: '1',
    examplesLength: 4
  };
```
(or even with a different variable name)

Reply:
Updated in https://github.com/elastic/kibana/pull/124710/commits/7634363c0ce8420f1ce649f17012731ecce30cae